packages/vite/src/modes/shadow-dom.ts

```diff
@@ -13,23 +13,43 @@ export function ShadowDomModuleModePlugin({ uno }: UnocssPluginContext): Plugin
     if (!element)
       return null
 
-    return useParts.filter(p => element[2].includes(p.rule)).map(({ rule, part }) => {
-      const idx = idxResolver(element[1])
-      return [
-        element[1],
-        `.${rule.replace(
-          /\[/g, '\\[',
-        ).replace(
-          /]/g, '\\]',
-        ).replace(
-          /:/g, '\\:')}::part(${part})`,
-        `${element[1]}:nth-of-type(${idx})::part(${part})`,
-      ]
-    })
+    const applyParts = useParts.filter(p => element[2].includes(p.rule))
+    if (applyParts.length === 0)
+      return null
+
+    const name = element[1]
+    return {
+      name,
+      entries: applyParts.map(({ rule, part }) => {
+        const idx = idxResolver(name)
+        return [
+          `.${rule.replace(
+            /\[/g, '\\[',
+          ).replace(
+            /]/g, '\\]',
+          ).replace(
+            /:/g, '\\:')}::part(${part})`,
```

Comment:
`rule.replace(/[:[\]]/g, '\\$&')`

As in this example:
https://github.com/antfu/unocss/blob/8c277dee9f411f6325332ccccbba6d1ddc16ac54/packages/core/src/utils/escape.ts#L2